mypy/checker.py

```diff
@@ -2483,8 +2483,16 @@ def check_assignment_to_multiple_lvalues(self, lvalues: List[Lvalue], rvalue: Ex
             # control in cases like: a, b = [int, str] where rhs would get
             # type List[object]
 
-            rvalues = rvalue.items
-
+            rvalues = []  # type: List[Expression]
+            for rval in rvalue.items:
+                if isinstance(rval, StarExpr):
+                    typs = get_proper_type(self.expr_checker.visit_star_expr(rval).type)
+                    if isinstance(typs, TupleType):
+                        rvalues.extend([TempNode(typ) for typ in typs.items])
+                    else:
+                        rvalues.append(TempNode(typs))
```

Comment:
If the type is an iterable, for example, the correct thing to do here would be to expand the right number of iterable item types, depending on the number of lvalues. It's okay if you don't want to implement that in this PR, however. The alternative would be to give an error about not being able to check the number of rvalues. The current behavior propagates an incorrect type, which I'd prefer to avoid.

Reply:
thanks for your review, i would like to implement the correct thing, so the basic idea is that just get the right number of iterable item types, instead of constructing a new `TempNode` type, right?

Reply:
The more correct thing would be figure out the number of rvalues needed here so that the number of rvalues matches the number of lvalues. Then assuming the rvalue type is an iterable, look at the iterable item type and append a suitable number of `TempNode` instances with the iterable item type. Here we can assume that the length of the iterable is correct, as we don't know how many items there will actually be.

There will be some interesting edge cases, I assume. If the type is not iterable and not a tuple, you can probably fall back to `AnyType` (make sure that an error gets reported either here or elsewhere if the rvalue type is bogus).